### ISSUE TITLE
Setup scraping mechanism

### ISSUE THREAD

**Issue by alexboii** [scrum-gang/jobhub-chrome#4]
Scraping tags:
 
- Company name
- Position
- Date posted
- URL
- _Deadlines (optional)_

**Comment by CamiloGarciaLaRotta** [comment]
After going through the API I see 2 ways of extracting information from the website:

- [Content Scripts](https://developer.chrome.com/extensions/content_scripts) \[[1](https://www.thepolyglotdeveloper.com/2018/09/creating-basic-chrome-extension/)] -> official solution
- Injecting a script to the current page which [sends](https://developer.chrome.com/extensions/runtime#method-sendMessage) DOM elements through messages between to background script \[[1](https://markb.co.uk/building-a-simple-google-chrome-extension.html)]\[[2](https://www.tomforth.co.uk/chromeextension/)] -> hacky solution

I opted for the Content Scripts approach.

--------------------------------

We do not want to be scrapping every website the user visits. It is both a waste of resources and may affect Chrome's speed (the user might uninstall the extension because it slows down browsing).

So we only want the scrapping to occur the moment the user clickc on the extension.

**Comment by CamiloGarciaLaRotta** [comment]
Useful reference: [List of all metatags](https://gist.github.com/lancejpollard/1978404)

**Comment by CamiloGarciaLaRotta** [comment]
Unfortunately, metatag names are very rarely common between websites. 
The closest I found to a consensus is the Open Graph tags, which Linkedin and Indeed don't implement.
I am working on the 3 following sites as reference: [[1](https://weworkremotely.com/remote-jobs/tigera-senior-software-engineer-golang)], [[2](https://www.linkedin.com/jobs/view/1087512462/)], [[3](https://www.indeed.ca/viewjob?jk=4b48b2bbf82266ec&tk=1d2iqufeb0gfs003&from=serp&vjs=3)]

The second, less elegant solution is to rely on RegEx to extract the fields we want directly from the body.

**Comment by alexboii** [comment]
```
let changeBackgroundColor = (maybeColor) =>
  /* See https://developer.chrome.com/extensions/tabs#method-executeScript.
     chrome.tabs.executeScript allows us to programmatically inject JavaScript
     into a page. Since we omit the optional first argument "tabId", the script
     is inserted into the active tab of the current window, which serves as the
     default. */
  switch maybeColor {
  | Some(color) =>
    let script = "document.body.style.backgroundColor=\"" ++ color ++ "\";";
    let scriptDetails = Tabs.mkScriptDetails(~code=script, ());
    Tabs.executeScript(scriptDetails)
  | None => ()
  };
```
taken from https://github.com/jchavarri/chrome-extension-starter-reason-react

**Comment by CamiloGarciaLaRotta** [comment]
I would vote against injection of code into the page to obtain DOM data.
Most injection approaches I found online were posted years before the introduction of Content Scripts.
Content Scripts were conceived exactly to avoid that pattern.

- Background scripts should be concerned with monitoring events.
- Popup scripts should be concerned with taking actions once the user has clicked on the extension icon.
- Content Scripts should be concerned with accessing and manipulating DOM of the visited websites.

They should all [communicate](https://developer.chrome.com/extensions/messaging#simple) through messages.

**Comment by CamiloGarciaLaRotta** [comment]
I have formatted this issue to be simply setting up the workflow for the scrapper.
The actual implementation of each scrapper unit will be handled by individual issues.
This issue was addressed by #11
